core/genesis.go

```diff
@@ -143,11 +148,17 @@ func (ga *GenesisAlloc) deriveHash() (common.Hash, error) {
 // flush is very similar with deriveHash, but the main difference is
 // all the generated states will be persisted into the given database.
 // Also, the genesis state specification will be flushed as well.
-func (ga *GenesisAlloc) flush(db ethdb.Database, triedb *trie.Database, blockhash common.Hash) error {
+func (ga *GenesisAlloc) flush(db ethdb.Database, triedb *trie.Database, blockhash common.Hash, cfg *params.ChainConfig) error {
 	statedb, err := state.New(types.EmptyRootHash, state.NewDatabaseWithNodeDB(db, triedb), nil)
 	if err != nil {
 		return err
 	}
+
+	// End the verkle conversion at genesis if the fork block is 0
+	if cfg != nil && cfg.IsCancun(big.NewInt(int64(0)), 0 /* XXX */) {
+		statedb.Database().EndVerkleTransition()
+	}
```

Comment:
@rjl493456442 I need to detect if verkle is enabled at genesis time, for testnets to work. I need to add the chain config as a parameter (and it's still not enough), but I think this isn't the right way to do it, would you have an opinion on how to best do this?